### PR TITLE
Possibility to send extra startup data (closes #2885)

### LINKED ISSUE #2885: Remove OPTION request for assignment phases.
same as #2862 . See [this function](https://github.com/OpenSlides/OpenSlides/blob/master/openslides/assignments/static/js/assignments/base.js#L319). This is called within some assignment resolvers.

### PR DESCRIPTION
**Pull request by FinnStutzenstein**

### CODE REVIEW

**Comment by FinnStutzenstein** [comment]
@normanjaeckel coverage failed. Do we have tests for autoupdate or the methods in the apps.py?

File: openslides/core/static/js/core/base.js [hunk 1]
```diff
@@ -294,6 +297,29 @@ angular.module('OpenSlidesApp.core', [
     }
 ])
 
+.factory('StartupExtraData', [
+    'Assignment',
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Please do not make 'Assignment' to be a dependecy of our core module. So solve this extraData thing we need another approach. Maybe something like we do for custom django settings. See https://github.com/OpenSlides/OpenSlides/blob/master/openslides/core/views.py#L181

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Please look at last commit. Now the phases are passed as an angular constant (in the wrong module ;) ). Now we have the dependency in the core/views.py accessing the assignment model to get the phases. Whats about this concept? (If its ok I'll clean up the code etc.)

**Comment by normanjaeckel** [comment]
No, I think also that the python module assignments should not be a dependency of core. So you can use a generic solution that loops over all apps (see [Django app registry](https://docs.djangoproject.com/en/1.10/ref/applications/#application-registry)) greps extra Angular constants (like we do for site and projector modules) and render this stuff into the response of the startup viewpoint.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Please review.

**Comment by normanjaeckel** [comment]
Maybe just add a test the checks if the custom variables and settings are part of the response of the api view call.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Added test.

**Comment by FinnStutzenstein** [comment]
@emanuelschuetze could you restart travis?

**Comment by emanuelschuetze** [comment]
Something seems to be wrong with travis connection here. I'll try to close/reopen...

**Comment by emanuelschuetze** [comment]
Travis done now.
@normanjaeckel Please review test.

**Pull request merged**